aws/resource_aws_eks_node_group.go

```diff
@@ -502,35 +503,35 @@ func resourceAwsEksNodeGroupUpdate(d *schema.ResourceData, meta interface{}) err
 		output, err := conn.UpdateNodegroupConfig(input)
 
 		if err != nil {
-			return fmt.Errorf("error updating EKS Node Group (%s) config: %w", d.Id(), err)
+			return diag.Errorf("error updating EKS Node Group (%s) config", d.Id(), err)
 		}
 
 		updateID := aws.StringValue(output.Update.Id)
 
-		_, err = waiter.NodegroupUpdateSuccessful(conn, clusterName, nodeGroupName, updateID, d.Timeout(schema.TimeoutUpdate))
+		_, err = waiter.NodegroupUpdateSuccessful(ctx, conn, clusterName, nodeGroupName, updateID, d.Timeout(schema.TimeoutUpdate))
 
 		if err != nil {
-			return fmt.Errorf("error waiting for EKS Node Group (%s) config update (%s): %w", d.Id(), updateID, err)
+			return diag.Errorf("error waiting for EKS Node Group (%s) config update (%s)", d.Id(), updateID, err)
 		}
 	}
 
 	if d.HasChange("tags_all") {
 		o, n := d.GetChange("tags_all")
 		if err := keyvaluetags.EksUpdateTags(conn, d.Get("arn").(string), o, n); err != nil {
-			return fmt.Errorf("error updating tags: %w", err)
+			return diag.Errorf("error updating tags", err)
```

Comment:
Still need to pass `error`, just can't use `%w`.

```suggestion
			return diag.Errorf("error updating tags: %s", err)
```